### PR TITLE
fix(editable):  input retaining focus when programmaticaly leaving input mode

### LINKED ISSUE #6819: Editable calls onSubmit when focusing the input on Firefox
### Description

After submitting the Editable by pressing `Enter`, it starts calling the `onSubmit` prop both when focusing and leaving the input. This issue doesn't occur on Chrome, or when submitting using a button. 

### Link to Reproduction

https://codesandbox.io/s/charming-cdn-sno833

### Steps to reproduce

On the CodeSandbox linked above, open the console and:

1. Click the Editable
2. Press Enter to submit, notice a console.log
3. Now click the Editable again, notice a console.log when focusing

That should not happen, the editable should be submitted when leaving the input or pressing the submit button, not when entering it. Following the steps above we get one log on Chrome, but two on Firefox.

### Chakra UI Version

2.3.5

### Browser

Firefox 105.0.3

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by anubra266**
Closes #6819 
Fixed input retaining focus when programmaticaly leaving input mode

### CODE REVIEW

File: packages/components/editable/stories/editable.stories.tsx [hunk 1]
```diff
@@ -84,16 +84,15 @@ const EditableControls = () => {
...
+    fontSize="2xl"
+    onSubmit={() => console.log("onSubmit", Math.random())}
+    isPreviewFocusable
   >
     <EditablePreview />
     <EditableInput />
-    <EditableControls />
+    {/* <EditableControls /> */}
   </Editable>
 )
```
Comment (segunadebayo) [inline packages/components/editable/stories/editable.stories.tsx:97]: Please revert these changes since they're unrelated

File: packages/components/editable/stories/editable.stories.tsx [hunk 2]
```diff
@@ -84,16 +84,15 @@ const EditableControls = () => {
...
-    onChange={console.log}
+    value="Rasengan ⚡️"
+    fontSize="2xl"
+    onSubmit={() => console.log("onSubmit", Math.random())}
+    isPreviewFocusable
   >
     <EditablePreview />
     <EditableInput />
-    <EditableControls />
+    {/* <EditableControls /> */}
```
Comment (anubra266) [inline packages/components/editable/stories/editable.stories.tsx:95]:
```suggestion
    defaultValue="Rasengan ⚡️"
    fontSize="xl"
    textAlign="center"
    isPreviewFocusable={false}
    submitOnBlur={false}
    onChange={console.log}
  >
    <EditablePreview />
    <EditableInput />
    <EditableControls />
```
